### ISSUE TITLE
Makefile.coq generates bad COQLIBINSTALL when DESTDIR set

### ISSUE THREAD

**Issue by wilbowma** [coq/coq#7233]
#### Version

The Coq Proof Assistant, version 8.7.2 (February 2018)
compiled on Feb 16 2018 8:50:49 with OCaml 4.06.0

#### Operating system

Arch Linux

#### Description of the problem

When `DESTDIR` is set, `concat_path` causes `COQLIBINSTALL` to equal `$(DESTDIR)/user-contrib`, instead of `$(DESTDIR)/$(COQLIB)/user-contrib`. My current workaround is to set `COQMF_WINDRIVE` to the empty string in `Makefile.coq.local`.

 I believe the problem is in the definition of `concat_path`, or in the function `windrive` in `coq_makefile`.

```makefile
concat_path = $(if $(1),$(1)/$(subst $(COQMF_WINDRIVE),/,$(2)),$(2))

COQLIBINSTALL = $(call concat_path,$(DESTDIR),$(COQLIB)user-contrib) 
```

`COQMF_WINDRIVE` is set to the same path (/usr/lib/coq) as `COQLIB` on my system (Arch Linux)  due to the definition of the `windrive` function in `coq_makefile`. So, `concat_path` removes `COQLIB` from the install path, when trying to remove `COQMF_WINDRIVE`.

**Comment by gares** [comment]
Could you please test #7241 ?

**Comment by wilbowma** [comment]
Sure

**Comment by wilbowma** [comment]
Couldn't test that branch directly; I was compiling mathcomp, which seems to be incompatible with 8.8. But, I applied the commit as a patch to 8.7.2 and recompiled mathcomp successfully, so it seems to work for me.

**Comment by Zimmi48** [comment]
Math-Comp is not supposed to be incompatible with 8.8...

**Comment by gares** [comment]
The 1.6.x release is... we are working on 1.7 but it will be released a bit after Coq 8.8

**Comment by gares** [comment]
@wilbowma Thanks!
